docs/standard/microservices-architecture/multi-container-microservice-net-applications/background-tasks-with-ihostedservice.md

```diff
@@ -250,9 +250,7 @@ The `IHostedService` interface provides a convenient way to start background tas
 
 -   **ASP.NET Core 2.1 Hosting samples** 
 
-    [*https://github.com/aspnet/Hosting/tree/dev/samples/GenericHostSample*](https://github.com/aspnet/Hosting/tree/dev/samples/GenericHostSample)
-
-
+    [*https://github.com/aspnet/Hosting/tree/release/2.1/samples/GenericHostSample*](https://github.com/aspnet/Hosting/tree/release/2.1/samples/GenericHostSample)
```

Comment:
I would recommend using a canonical link instead. This original link was broken because we deleted the dev branch. We might delete release/2.1 at some point.

https://github.com/aspnet/Hosting/tree/1596c2ba5508fceefd929228b4e5e95e50b1d4d4/samples/GenericHostSample

Reply:
That's a pretty ugly URL to show to customers given the whole URL is printed in these e-books 😄 

If you get to the point of deleting that branch, it would probably be time for reviewing this topic no?

Reply:
Possibly, but we're never going to remember to do that when/if we delete branches. It may be better to just copy that sample into the docs instead of relying on it existing on GitHub.

 Or, use a url shortener that produces a less-ugly URL.

Reply:
That would be on us to run broken link tests and not on you. If the target looks fine, I'll keep it as is. We're trying to avoid url shorteners as much as possible, since the community cannot edit them.